main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala

```diff
@@ -76,6 +84,106 @@ object GenNamespaces {
+                                 defn: Def,
+                                 sym: Symbol.DefnSym,
+                                 nsClassType: JvmType.Reference,
+                                 ifoType: JvmType.Reference,
+                                 ns: NamespaceInfo)(implicit root: Root, flix: Flix): Unit = {
+    // Name of the shill
+    val name = JvmOps.getDefMethodNameInNamespaceClass(sym)
+
+    // Jvm type of method args
+    val args = defn.tpe.typeArguments.map(JvmOps.getErasedType)
+
+    // Length of args in local
+    val varLen = args.init.map{
+      case JvmType.PrimLong | JvmType.PrimDouble => 2
+      case _ => 1
+    }.sum
+
+
+    // Method header
+    val method = visitor.visitMethod(ACC_PUBLIC + ACC_FINAL + ACC_STATIC, name, AsmOps.getMethodDescriptor(args.init, args.last), null, null)
+    method.visitCode()
+
+    // Creating a context object
+    method.visitTypeInsn(NEW, JvmName.Context.toInternalName)
+    method.visitInsn(DUP)
+
+    // Calling the constructor of context object
+    method.visitMethodInsn(INVOKESPECIAL, JvmName.Context.toInternalName, "<init>",
+      AsmOps.getMethodDescriptor(Nil, JvmType.Void), false)
+
+
+    // creating a local variable for context
+    val label = new Label
+    method.visitLabel(label)
+    method.visitLocalVariable("context", JvmType.Context.toDescriptor, null, label, label, varLen)
+
+    // Putting another reference of context on top of the stack
+    method.visitInsn(DUP)
+
+    // Storing Context on the variable
+    method.visitVarInsn(ASTORE, varLen)
+
+    // Name of the field for namespace on context object
+    val nsFieldName = JvmOps.getNamespaceFieldNameInContextClass(ns)
+
+    // Name of the field for IFO of defn on namespace
+    val defnFieldName = JvmOps.getDefFieldNameInNamespaceClass(sym)
+
+    // Extracting the namespace field from the context object
+    method.visitFieldInsn(GETFIELD, JvmName.Context.toInternalName, nsFieldName, nsClassType.toDescriptor)
+
+    // Extracting the ifo from namespace
+    method.visitFieldInsn(GETFIELD, nsClassType.name.toInternalName, defnFieldName, ifoType.toDescriptor)
+
+    // Offset for each parameter
+    var offset: Int = 0
+
+    // Set arguments for the IFO
+    for((arg, index) <- args.init.zipWithIndex) {
+      // Duplicate the IFO reference
+      method.visitInsn(DUP)
+
+      // Get the argument from the field
+      val iLoad = AsmOps.getLoadInstruction(arg)
+      method.visitVarInsn(iLoad, offset)
+
+      // Call the setter for the argument
+      method.visitMethodInsn(INVOKEVIRTUAL, ifoType.name.toInternalName, s"setArg$index",
+        AsmOps.getMethodDescriptor(List(arg), JvmType.Void), false)
+
+      // Incrementing the offset
+      arg match {
+        case JvmType.PrimLong | JvmType.PrimDouble => offset += 2
+        case _ => offset += 1
+      }
+    }
+
+    // TODO: Replace this with an appropriate while loop, for now we just call the apply
```

Comment:
Can we do that before merging in, or not?

Reply:
I rather wait until we want to do `apply` expressions in `compileExp`.